rootston/desktop.c

```diff
@@ -187,6 +188,32 @@ static struct wlr_output *view_get_output(struct roots_view *view) {
 		output_y);
 }
 
+void view_arrange_maximized(struct roots_view *view) {
+	struct wlr_box view_box;
+	view_get_box(view, &view_box);
+
+	view->maximized = true;
+	view->saved.x = view->x;
+	view->saved.y = view->y;
+	view->saved.rotation = view->rotation;
+	view->saved.width = view_box.width;
+	view->saved.height = view_box.height;
```

Comment:
We only want to save the position + size + rotation of the view if it isn't yet maximized. This function can be called when the view is already maximized from the layer surface code.

Reply:
Oh that explains the problem I was hitting earlier